Makefile-utils.mk

```diff
@@ -0,0 +1,104 @@
+# Portability
+ifdef ComSpec
+	RM := del /F /Q
+    SLASH :=\\
+else
+	RM := rm -rf
+    SLASH :=/
+endif
+
+# Use $/ as the path separator
+/ := $(strip $(SLASH))
+
+# File classes
+FILES_JS := $(shell \
+	find . \
+		-not \( -path '.$/.git' -prune \) \
+		-not \( -path '.$/build' -prune \) \
+		-not \( -path '.$/node_modules' -prune \) \
+		-not \( -path '.$/public' -prune \) \
+		-type f \
+		\( -name '*.js' -or -name '*.jsx' \) \
+)
+
+FILES_SCSS := $(shell \
+	find client assets \
+		-type f \
+		-name '*.scss' \
+)
+
+# git
+GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+
+# node/npm helpers
+NODE_ENV ?= development
+NPM_BIN  := node_modules$/.bin$/
+NODE     := $(shell which node)
+NPM      := $(shell which npm)
+
+# $(call npm-deps,package-a package-b package-c)
+npm-deps = $(addprefix node_modules$/,$1) missing-modules
+
+.PHONY: missing-modules
+missing-modules:
+	$(RM) .make-cache$/missing-modules; \
+  # Due to a bug in npm it's removing packages on install and
```

Comment:
A nitpick, but it seems weird that these are indented with spaces and not tabs, unless I'm so out of Makefile practice that they can't be tabs?

Reply:
these are `make` comments but if they were indented with a tab they would be `bash` comments and would actually execute when building the target

it's odd and the author of `make` has some funny bits to say about it, but if it's indented with a tab it's a command and if it's indented with spaces it's a `make` directive. odd but once you learn that it's not confusing.

Reply:
Ah okay, so I'm indeed out of practice 😅